cli/data.go

```diff
@@ -68,23 +68,35 @@ func (c *viamClient) dataExportAction(cCtx *cli.Context) error {
 
 // DataDeleteAction is the corresponding action for 'data delete'.
 func DataDeleteAction(c *cli.Context) error {
-	filter, err := createDataFilter(c)
-	if err != nil {
-		return err
-	}
-
 	client, err := newViamClient(c)
 	if err != nil {
 		return err
 	}
 
 	switch c.String(dataFlagDataType) {
 	case dataTypeBinary:
+		filter, err := createDataFilter(c)
+		if err != nil {
+			return err
+		}
 		if err := client.deleteBinaryData(filter); err != nil {
 			return err
 		}
 	case dataTypeTabular:
-		if err := client.deleteTabularData(filter); err != nil {
+		orgIDs := c.StringSlice(dataFlagOrgIDs)
+		if orgIDs == nil || len(orgIDs) != 1 {
+			return errors.New("must provide single org id")
+		}
+
+		if !c.IsSet(dataFlagDeleteTabularDataOlderThanDays) {
+			return errors.Errorf(
+				"must set %s flag, set to 0 to delete all tabular data in the org. "+
```

Comment:
We should probably extract out 'delete-binary' and 'delete-tabular' into two separate actions, but that involves documentation changes to be coordinated with this.

Wanted to get this in before the release so that we could finally fully remove DeleteTabularDataByFilter from the API without any stable RDK or app dependencies on it.